google/cloud/storage/client_sign_policy_document_test.cc

```diff
@@ -0,0 +1,106 @@
+      "client_email": "foo-email@foo-project.iam.gserviceaccount.com",
+      "client_id": "100000000000000000001",
+      "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+      "token_uri": "https://oauth2.googleapis.com/token",
+      "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+      "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/foo-email%40foo-project.iam.gserviceaccount.com"
+})""";
+
+PolicyDocument CreatePolicyDocumentForTest() {
+  PolicyDocument result;
+  result.expiration_time = internal::ParseRfc3339("2010-06-16T11:11:11Z");
+  result.conditions.emplace_back(
+      PolicyDocumentCondition::StartsWith("key", ""));
+  result.conditions.emplace_back(
+      PolicyDocumentCondition::ExactMatchObject("acl", "bucket-owner-read"));
+  result.conditions.emplace_back(
+      PolicyDocumentCondition::ExactMatchObject("bucket", "travel-maps"));
+  result.conditions.emplace_back(
+      PolicyDocumentCondition::ExactMatch("Content-Type", "image/jpeg"));
+  result.conditions.emplace_back(
+      PolicyDocumentCondition::ContentLengthRange(0, 1000000));
+  return result;
+}
+
+TEST(SignedPolicyDocumentIntegrationTest, Sign) {
+  auto creds = oauth2::CreateServiceAccountCredentialsFromJsonContents(
+      kJsonKeyfileContents);
+  ASSERT_STATUS_OK(creds);
+  Client client(*creds);
+
+  auto actual = client.CreateSignedPolicyDocument(
+      "test-bucket", CreatePolicyDocumentForTest());
+  ASSERT_STATUS_OK(actual);
+
+  EXPECT_THAT(*actual, HasSubstr("test-bucket"));
+
+  std::string expected =
+      "https://storage.googleapis.com/test-bucket"
+      "?GoogleAccessId=foo-email@foo-project.iam.gserviceaccount.com"
+      "?Expires=1276686671"
+      "?Policy="
+      "eyJjb25kaXRpb25zIjpbWyJzdGFydHMtd2l0aCIsIiRrZXkiLCIiXSx7ImFjbCI6ImJ1Y2tl"
+      "dC1vd25lci1yZWFkIn0seyJidWNrZXQiOiJ0cmF2ZWwtbWFwcyJ9LFsiZXEiLCIkQ29udGVu"
+      "dC1UeXBlIiwiaW1hZ2UvanBlZyJdLFsiY29udGVudC1sZW5ndGgtcmFuZ2UiLCIwIiwiMTAw"
+      "MDAwMCJdXSwiZXhwaXJhdGlvbiI6IjIwMTAtMDYtMTZUMTE6MTE6MTFaIn0K"
+      "?Signature=pYCvNvqPjMC6pZTR8x9rraHDniaHqez/5XR8Cr0UHdAiDico7Sqw6kw9Li/"
```

Comment:
The policy matches but the signature doesn't. I'm using the example [python script](https://cloud.google.com/storage/docs/xml-api/post-object#policydocument) to generate the signature to match against. I also tried using `openssl dgst -sha256 -sign private.pem blob.txt | openssl base64 -A` and got the same result. Incidentally I can't reproduce the example here:

https://github.com/googleapis/google-cloud-cpp/blob/master/google/cloud/storage/oauth2/service_account_credentials_test.cc#L382

So maybe something is wrong with how I'm doing things and not the code.